### DISCUSSION TITLE
Re-License of Vaultwarden to AGPLv3

### DISCUSSION THREAD

**Discussion by BlackDex** [dani-garcia/vaultwarden#2450] [Ideas]
**Hello contributors to Vaultwarden,**

I'm starting this discussion because we want to change the open source license of Vaultwarden.
Currently it is licensed under GPLv3 which is nice, but this has a gap which could allow commercial usage of Vaultwarden without providing back to the community.
For this reason Bitwarden has it's server licensed under the AGPLv3. That same license we want to use for Vaultwarden to keep Vaultwarden fully open source.

To be clear, this will not effect any (end)user or any self-hosted environment which you share with your family and friends.
This will only have an effect on companies or self-hosted environments that build proprietary tools using Vaultwarden code or modify the Vaultwarden code.
This is also more fair towards Bitwarden, since currently the GPLv3 license of Vaultwarden could allow other companies to compete with Bitwarden by modifying the Vaultwarden code and not providing these changes back to the community, which is not something we want.

## What we need to do

We need a consent from all major contributors who made Vaultwarden towards what it is now. This only includes actual changes to the Rust code and does not include spelling changes, formatting, clippy fixes etc...
For this we have compiled a list of all these users from whom we would like to have a consent to continue to fully change the Vaultwarden code to AGPLv3.

The only thing you need to do is comment below using the following template, and change the parts in between `{{ }}`:

```
I, {{ Name }} hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by {{ github username }} are copyright of {{ name }}.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.
```

If for some reason you do not want your real name to be visible, please use the same as is visible on Github, or just your Github username instead.

## What is GPL?

[GPL (Gnu Public License)](https://www.gnu.org/licenses/gpl-3.0.html) was originally written by Richard Stallman, the founder of the Free Software Foundation for the GNU project. All licenses in the GPL series are copyleft licenses which means that the derivative works should be released under the same license or an equivalent license.

## The difference between GPL and AGPL

GPL says that if the software is redistributed, you have to release the source code of the modification. But the GPL license doesn't mention anything about modifying the software and providing it as a service. This is due to the fact that letting users interact with software over a network cannot be considered as redistribution. Redistribution is thus the triggering event of GPL. This is often referred to as the SaaS loophole of GPL.

AGPL added one more requirement to GPL, which is:

> If you run a modified program on a server and let other users communicate with it there, your server must also allow them to download the source code corresponding to the modified version running there.

This just means that the software cannot be sold as a proprietary product with modification without providing the source of the modifications to the users of the product.

[AGPL](https://www.gnu.org/licenses/licenses.html#AGPL) Is also recommended by the Free Software Foundation (FSF) if the software is commonly run over a network, which is what Vaultwarden is.

## Tagging contributors

This list is just in alphabetic order. I hope i haven't missed anyone, if I did, please let us know, but it would be even better if you would sign it right away :smile: .

:heavy_check_mark: @aksdb
:heavy_check_mark: @ArmaanT
:heavy_check_mark: @aveao
:heavy_check_mark: @Baelyk
:heavy_check_mark: @BlackDex
:heavy_check_mark: @dani-garcia
:heavy_check_mark: @domdomegg
:heavy_check_mark: @dongcarl
:heavy_check_mark: @endyman
:heavy_check_mark: @farodin91
:heavy_check_mark: @janost
:heavy_check_mark: @jjlin
:heavy_check_mark: @Kurnihil
:heavy_check_mark: @MFijak
:heavy_check_mark: @mprasil
:heavy_check_mark: @njfox
:heavy_check_mark: @olivierIllogika
:heavy_check_mark: @paolobarbolini
:heavy_check_mark: @pjsier 
:heavy_check_mark: @RealOrangeOne
:heavy_check_mark: @rfwatson
:heavy_check_mark: @samueltardieu
:heavy_check_mark: @shauder
:heavy_check_mark: @sirux88
:heavy_check_mark: @soruh
:heavy_check_mark: @Skeen
:heavy_check_mark: @stefan0xC
:heavy_check_mark: @Step7750
:heavy_check_mark: @swedishborgie
:heavy_check_mark: @tessus
:heavy_check_mark: @thelittlefireman
:heavy_check_mark: @TheMardy
:heavy_check_mark: @TinfoilSubmarine
:heavy_check_mark: @tomuta
:heavy_check_mark: @tycho
:heavy_check_mark: @ViViDboarder
:heavy_check_mark: @vplme

## Thanks

And, thanks for all your support to the Vaultwarden Project!

**Comment by BlackDex** [comment]
I, Mathijs van Veluw hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by BlackDex are copyright of Mathijs van Veluw.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by TinfoilSubmarine** [comment]
I, Joel Beckmeyer hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by TinfoilSubmarine are copyright of Joel Beckmeyer.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by aveao** [comment]
I, Ave Özkal hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by aveao are copyright of Ave Özkal.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by ViViDboarder** [comment]
I, Ian hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by ViViDboarder are copyright of Ian.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by TheMardy** [comment]
I, Martijn Hanegraaf hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by TheMardy are copyright of Martijn Hanegraaf.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by swedishborgie** [comment]
I, Michael Powers hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by swedishborgie are copyright of Michael Powers.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by ArmaanT** [comment]
I, Armaan Tobaccowalla hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by ArmaanT are copyright of Armaan Tobaccowalla.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by mprasil** [comment]
I, Miro Prasil hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by mprasil are copyright of Miro Prasil.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by shauder** [comment]
I, Shane Faulkner hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by shauder are copyright of Shane Faulkner.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by rfwatson** [comment]
I, Rob Watson, hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @rfwatson are copyright of Rob Watson.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by dani-garcia** [comment]
I, Daniel García hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @dani-garcia are copyright of Daniel García.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by janost** [comment]
I, Janos Tassy hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @janost are copyright of Janos Tassy.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by jjlin** [comment]
While I don't object to changing the license, I think a few things mentioned in the intro are not quite true and should be clarified.

> To be clear, this will not effect any (end)user or any self-hosted environment which you share with your family and friends.

If you make your Vaultwarden instance publicly accessible, then there is a reasonable expectation that anyone could interact  with it (if only to load the login page), so if you've modified any of the Vaultwarden backend (e.g., custom templates or graphics that get built into the binary), I believe you would technically be obligated to "[prominently offer](https://www.gnu.org/licenses/agpl-3.0.html#section13)" the source. In practice, that would probably mean modifying the web vault login page with a link to your source.

> This will only have an effect on companies that build proprietary tools using Vaultwarden code.

As mentioned above, this also affects non-commercial users who have modified the Vaultwarden code.

> This is also more fair towards Bitwarden, since currently the GPLv3 license of Vaultwarden could allow other companies to compete with Bitwarden, which is not something we want.

While this may prevent deep integration of Vaultwarden with a company's other products, they are still perfectly able to provide standalone instances, even with some customizations (as long as they make the modified source available), so I don't think this change would have much effect on most MSPs who offer managed Vaultwarden services.

Overall, AGPL has a limited ability to prevent third-party competition with the original product, as can be seen with Bitwarden's shift towards a non-open-source ("source-available") license for key portions of their software.

**Reply by dumblob** [reply]
I'd add that the following sentence:

>This just means that the software cannot be sold as a proprietary product.

is also not quite true. AFAIK AGPL does in **no way** forbid selling (or otherwise profiting from) a modified version of the SW. It just requires you to provide this very particular running SW piece also in source form (only) to those "communicationg with it" (in case of Vaultvarden you can thus skip anyone not having a working login credentials as they can not really "communicate with it" in any sensible way).

**Reply by BlackDex** [reply]
Thanks for the input @jjlin and @dumblob, i have updated the message to better reflect this.

**Comment by jjlin** [comment]
I, Jeremy Lin, hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @jjlin are copyright of Jeremy Lin.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by domdomegg** [comment]
I, Adam Jones hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by domdomegg are copyright of Adam Jones.

And I hereby grant the Vaultwarden project team to re-license Vaultwarden, including all past, present and future contributions of the author listed above.

**Comment by aksdb** [comment]
I, Andreas Schneider hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @aksdb are copyright of Andreas Schneider.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by endyman** [comment]
I, Nils Domrose hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @endyman are copyright of Nils Domrose.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by Skeen** [comment]
I, Emil Madsen hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @Skeen are copyright of Emil Madsen.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by RealOrangeOne** [comment]
I, Jake Howard hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @RealOrangeOne  are copyright of Jake Howard.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by Baelyk** [comment]
I, George Ekman hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @Baelyk are copyright of George Ekman.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by tycho** [comment]
I, Steven Noonan hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @tycho are copyright of Steven Noonan.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by paolobarbolini** [comment]
I, Paolo Barbolini hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @paolobarbolini are copyright of Paolo Barbolini.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by BlackDex** [comment]
I want to thank everybody who signed this already! Thanks for your support!

There are still a few people who contributed from whom we would like a comment also.
We would really appreciate it if you could spare a little bit of your time to approve the re-licenese with your comment.

Thanks in advance!!
Re-Mention:
@dongcarl
@endyman
@krankur
@njfox
@olivierIllogika
@Step7750
@thelittlefireman
@tomuta
@vplme

**Comment by dongcarl** [comment]
I, Carl Dong hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @dongcarl are copyright of Carl Dong.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by njfox** [comment]
I, Nick Fox hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @njfox are copyright of Nick Fox.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by BlackDex** [comment]
Time to bump again. There are only a few people left.
We would really appreciate it if you could spare a little bit of your time to approve the re-licenese with your comment.

Also, if you are against this change, let us know that too, and why. 

@endyman
@krankur
@olivierIllogika
@Step7750
@thelittlefireman
@tomuta
@vplme

And thanks!

**Comment by thelittlefireman** [comment]
I, under the alias thelittlefireman grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @thelittlefireman are copyright of thelittlefireman
And i grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by endyman** [comment]
I, Nils Domrose hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @endyman are copyright of Nils Domrose.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by o-kotb** [comment]
Is it possible to do something like how [the dolphin emulator team did when they relicensed their project?](https://dolphin-emu.org/blog/2015/05/25/relicensing-dolphin/) If it's 95% of contributions, then you might be able to relicense vaultwarden faster.

**Comment by BlackDex** [comment]
Were are so close to all people having this signed! Thanks for everybody who did this already!

Just a few people left, would be great if you could spare just about 2 minutes if not less to sign this, that would be very appreciated.

Thanks in advance.

@krankur
@olivierIllogika
@Step7750
@tomuta
@vplme

**Comment by stefan0xC** [comment]
I, Stefan Melmuk hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @stefan0xC are copyright of Stefan Melmuk.
And I hereby grant the Vaultwarden project team the permission to re-license Vaultwarden under the GNU Affero General Public License, including all past, present and future contributions of the author listed above.

**Comment by bt90** [comment]
@BlackDex some of those missing have their mail address listed in their Github profile.

**Reply by BlackDex** [reply]
They should receive the mentions also.

**Reply by bt90** [reply]
Depends on their notification settings and/or if they visit Github regularly.

**Reply by bt90** [reply]
Mailed them :)

**Comment by vplme** [comment]
I, Victor Verstappen hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @vplme are copyright of Victor Verstappen.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by Step7750** [comment]
I, Stepan Fedorko-Bartos hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @Step7750  are copyright of Stepan Fedorko-Bartos.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Reply by BlackDex** [reply]
Thanks @Step7750!
Looks like you missed a few replacement fields, name and GitHub username.

**Reply by Step7750** [reply]
Done @BlackDex

**Comment by bt90** [comment]
@BlackDex the remaining contributors should all have received an email from me.

**Comment by olivierIllogika** [comment]
I, Olivier Martin hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @olivierIllogika are copyright of Olivier Martin.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by stefan0xC** [comment]
I've sent a mail to @tomuta and @krankur as well. And I think that @samueltardieu @MFijak @manofthepeace @am97 @pjsier and @tessus made some relevant changes in the meantime too, but I might have missed someone? (Not sure what exactly counts as major contribution.) Could you add them to your list?

And I think the people with a promising open pull request (like @sirux88 @farodin91 and @Kurnihil) could also agree to a possible Re-licensing of their contribution in advance? Or would that be too premature. :sweat_smile:

**Reply by kpfleming** [reply]
For safety it would probably be best to require all unmerged PRs, and all future PRs, to be dual-licensed under the old and new licenses. That way the rest of code base can move to the new license when the agreements have been provided.

**Reply by BlackDex** [reply]
> For safety it would probably be best to require all unmerged PRs, and all future PRs, to be dual-licensed under the old and new licenses.

That isn't possible, they aren't compatible. AGPLv3 rules out GPLv3.

**Reply by stefan0xC** [reply]
I am also not sure if this is necessary. I mean if the as of yet undecided contributors all agree and we officially switch the codebase to AGPLv3 then we could require the authors of the then remaining unmerged PRs to agree as part of the process to get it merged (or live without their contributions) but I think asking them in advance if they would be okay with it is probably nicer (and if they happen to be against it we could find out/discuss why?)

**Reply by BlackDex** [reply]
If they rebase, it's official ;)

**Reply by tessus** [reply]
I doubt that my contributions were anything close to major, but nevertheless I'll add my agreement notice below. I certainly doesn't hurt, even if it is unnecessary.

**Reply by dancojocaru2000** [reply]
> That isn't possible, they aren't compatible. AGPLv3 rules out GPLv3.

It totally is possible. Licenses don't exclude each other. See the way QT does it: you can either get their software GPL-licensed, or pay for a commercial license.

And, if they dual-license, then the project can switch to AGPLv3. Even if small parts of the project are licensed GPLv3 as well, someone who wants to use the projects GPLv3 part would have to rewrite everything else.

**Comment by samueltardieu** [comment]
I, Samuel Tardieu hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @samueltardieu are copyright of Samuel Tardieu.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by tessus** [comment]
I, Helmut K. C. Tessarek hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @tessus are copyright of Helmut K. C. Tessarek.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by pjsier** [comment]
I, Pat Sier hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @pjsier are copyright of Pat Sier.
And I hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by MFijak** [comment]
I, Maximilian Fijak hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @MFijak are copyright of Maximilian Fijak.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by tomuta** [comment]
I, @tomuta hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @tomuta are copyright of @tomuta.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by theoparis** [comment]
Why not MIT or BSD license instead of GPL? 😕

**Reply by bt90** [reply]
This is because both MIT and BSD allow a commercial product to be derived from Vaultwarden whose source code would not have to be disclosed. And the reasoning behind that is already explained in the initial post of the thread.

**Reply by stefan0xC** [reply]
Because MIT/BSD licenses are not copyleft.
See also the first post for the reasons why this matters (i.e. that the Bitwarden server itself uses the AGPLv3) or why we want to switch to AGPLv3 in particular.

**Comment by Kurnihil** [comment]
I, Daniele Andrei hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by Kurnihil are copyright of Daniele Andrei.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by sirux88** [comment]
I, Matthias G. hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @sirux88 are copyright of Matthias G.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by farodin91** [comment]
I, Jan Jansen hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @farodin91 are copyright of Jan Jansen.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by soruh** [comment]
I, Paul Römer hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by @soruh are copyright of Paul Römer.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.

**Comment by soruh** [comment]
I think the comment is missing a word btw. It should probably say:
```patch
-And i hereby grant the Vaultwarden project team to re-license Vaultwarden
+And I hereby grant the Vaultwarden project team permission to re-license Vaultwarden
```
right?

**Reply by tessus** [reply]
I don't think it matters much. We all just copy and pasted the text from the first post, so I doubt that everyone will add those 2 words now. ;-)

**Reply by soruh** [reply]
Yeah, I just thought it might make sense to add the word to the first post for the future, but IANAL so I don't know how important that is.

**Reply by tessus** [reply]
not anymore, since now the project is already AGPL. Thus this does not have to be done anymore.

This was just necessary to re-license the already contributed code....

**Reply by BlackDex** [reply]
No, and is not needed. It's a text out of your name, like a oath. I never saw text or heard someone use and.

**Comment by svin24** [comment]
```
I, {{ Name }} hereby grant permission to re-license Vaultwarden under the GNU Affero General Public License.

A portion of the commits made by {{ github username }} are copyright of {{ name }}.
And i hereby grant the Vaultwarden project team to re-license Vaultwarden,
including all past, present and future contributions of the author listed above.
```

Small question as i am curious: How much is this necessary from a transition to GPL to AGPL. because this sounds like a soft form of CLA.

**Reply by BlackDex** [reply]
For one, Vaultwarden is at this point AGPLv3 now, so it's no longer needed to sign this.
Further, most if not all software in the Open Source world has some license, and if you intend to work on that software you need to comply to the license which that code is using.

Because we wanted to transition from GPLv3 to AGPLv3, we needed consent from previous and new contributors. Since this is done all new contributions from anyone automatically consent with the AGPLv3 license, since that is what we currently use.

**Reply by bt90** [reply]
I think the question was more about the phrase being written in the form of a "blank cheque", in the sense of "we're free relicense to whatever we want in the future".

**Reply by BlackDex** [reply]
No we are not, since it states ` grant permission to re-license Vaultwarden under the GNU Affero General Public License`.
It specifically states the AGPL and nothing else. So, no we can not change it again to EUPL when we want for example.

**Reply by svin24** [reply]
> No we are not, since it states ` grant permission to re-license Vaultwarden under the GNU Affero General Public License`. It specifically states the AGPL and nothing else. So, no we can not change it again to EUPL when we want for example.

Thank you for the clarification.

**Comment by BlackDex** [comment]
I want to thank everybody who signed the consent to make it possible to convert Vaultwarden from GPLv3 to AGPLv3 🥳 
We couldn't have done this without you. Not only the Re-License, but also bringing Vaultwarden to the state it currently is at.
We hope you keep contributing and help make Vaultwarden better.

Thanks!

**Comment by BlackDex** [comment]
To save the current state I have created a PDF of this topic and attached it to this post.

[Re-License of Vaultwarden to AGPLv3 - Discussion #2450 - dani-garcia_vaultwarden - 2023-02-13.pdf](https://github.com/dani-garcia/vaultwarden/files/10723635/Re-License.of.Vaultwarden.to.AGPLv3.-.Discussion.2450.-.dani-garcia_vaultwarden.-.2023-02-13.pdf)

**Reply by williamdes** [reply]
you should probably lock this discussion now or in some time

**Reply by BlackDex** [reply]
Indeed. I will do that right away.
